telethon/client/chats.py

```diff
@@ -762,4 +762,229 @@ def action(
+
+        Unless otherwise stated, permissions will work in channels and megagroups.
+
+        Arguments
+            entity (`entity`):
+                The channel, megagroup or chat where the promotion should happen.
+
+            user (`entity`):
+                The user to be promoted.
+
+            change_info (`bool`, optional):
+                Whether the user will be able to change info.
+
+            post_messages (`bool`, optional):
+                Whether the user will be able to post in the channel.
+                This will only work in broadcast channels.
+
+            edit_messages (`bool`, optional):
+                Whether the user will be able to edit messages in the channel.
+                This will only work in broadcast channels.
+
+            delete_messages (`bool`, optional):
+                Whether the user will be able to delete messages.
+
+            ban_users (`bool`, optional):
+                Whether the user will be able to ban users.
+
+            invite_users (`bool`, optional):
+                Whether the user will be able to invite users. Needs some testing.
+
+            pin_messages (`bool`, optional):
+                Whether the user will be able to pin messages.
+
+            add_admins (`bool`, optional):
+                Whether the user will be able to add admins.
+
+            is_admin (`bool`, optional):
+                Whether the user will be an admin in the chat.
+                This will only work in small group chats.
+
+        Returns
+            The resulting :tl:`Updates` object.
+
+        Example
+            .. code-block:: python
+
+                # Allowing `user` to pin messages in `chat`
+                client.edit_admin(chat, user, pin_messages=True)
+        """
+        entity = await self.get_input_entity(entity)
+        user = await self.get_input_entity(user)
+        if not isinstance(user, types.InputPeerUser):
+            raise ValueError('You must pass a user entity')
+
+        if isinstance(entity, types.InputPeerChannel):
+            return await self(functions.channels.EditAdminRequest(
+                entity,
+                user,
+                types.ChatAdminRights(
+                    change_info=change_info,
+                    post_messages=post_messages,
+                    edit_messages=edit_messages,
+                    delete_messages=delete_messages,
+                    ban_users=ban_users,
+                    invite_users=invite_users,
+                    pin_messages=pin_messages,
+                    add_admins=add_admins
+                )
+            ))
+        elif isinstance(entity, types.InputPeerChat):
+            return await self(functions.messages.EditChatAdminRequest(entity, user, is_admin=is_admin))
+        else:
+            raise ValueError('You must pass either a channel or a supergroup or a normal group')
+
+    async def edit_permission(
```

Comment:
Should this be renamed as `edit_restriction`? Admin permissions makes more sense, and that's `edit_admin`.